iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/converter/MetadataConverterUtils.java

```diff
@@ -60,5 +61,31 @@ private static Year parseYearOrGetNull(String year, Logger log) {
             return null;
         }
     }
-    
+
+    /**
+     * Returns pid list constructed from a {@link Result}. Pid list is constructed from {@link Instance#pid} fields if they're not empty.
+     * Otherwise it's constructed from {@link Instance#alternateIdentifier} fields. In any other case an empty list is
+     * returned.
+     */
+    public static List<StructuredProperty> extractPid(Result result) {
+        if (Objects.isNull(result.getInstance())) {
+            return Collections.emptyList();
+        }
+
+        List<StructuredProperty> pid = result.getInstance().stream()
+                .map(x -> Optional.ofNullable(x.getPid()).map(Collection::stream))
+                .filter(Optional::isPresent)
+                .flatMap(Optional::get)
+                .collect(Collectors.toList());
+
+        if (CollectionUtils.isNotEmpty(pid)) {
+            return pid;
+        }
```

Comment:
We should probably get as many identifiers as possible, from both sources (namely `instance#pid` and `instance#alternateIdentifier`). So we should not return results after inspecting the first source of ids only.

The logic I was describing in the main ticket ("treat `Result#instance#pid` as the more trustable than `Instance#alternateIdentifiers` just to make sure we have a specific non-random rule of picking the ids") was referring to dealing with duplicates whenever the same type of id occur in both places and we end up with an overlap. Sorry I didn't express that clearly enough.

When I was writing the #1272 description **I had the map type in mind** (where the duplicates resolution is a must) but since we are building a `List<StructuredProperty>` we are not forced to deal with duplicates, technically, so maybe we can just trust the disjoint sets and simply propagate the ids from both fields without the duplicates resolution.

Reply:
Ok, I got mislead by the slack conversation also. Fixed this, I've also added duplicates resolution, just to be on the safe side, and also because it was quite easy to do it with the a list.